### ISSUE TITLE
build_targets Index Out Of Bounds Error

### ISSUE THREAD

**Issue by ganLover** [eriklindernoren/PyTorch-YOLOv3#131]
While training the system on custom dataset, [x_center, y_center, width, height] (format of the ground truth- wrt. the image) eg- (93,100,60,70)

In the build_targets():

            gx = target[b, t, 1] * nG
            gy = target[b, t, 2] * nG
            gw = target[b, t, 3] * nG
            gh = target[b, t, 4] * nG
            # Get grid box indices
            gi = int(gx)
            gj = int(gy)

.......

conf_mask[b, anch_ious > ignore_thres, gj, gi] = 0

As per example -> 
gx = 93 * 13
gi = 93 * 13

But -> conf_mask = torch.ones(nB, nA, nG, nG)  [Batch_size, no_of_anchors, grid_size, grid_size]
Where nG = 13

That is why I am getting index out of bound error.
If I am wrong pls explain what is the format of Target labels.

**Comment by fourth-archive** [comment]
Try this repo: https://github.com/ultralytics/yolov3

It comes with a tutorial that answers your format questions:
https://github.com/ultralytics/yolov3/wiki/Train-Custom-Data

**Comment by ganLover** [comment]
> Try this repo: https://github.com/ultralytics/yolov3
> 
> It comes with a tutorial that answers your format questions:
> https://github.com/ultralytics/yolov3/wiki/Train-Custom-Data

Thanks

**Comment by eriklindernoren** [comment]
@ganLover `x`, `y` (bbox center) in the target data are normalized [here](https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/utils/datasets.py#L113), and will be in the range `[0, 1]` in `build_targets`, so we multiply it with the grid size to find out in what grid the bounding box center is positioned, e.g. if `x`, `y` is `[0.5 0.5]` then `[0.5, 0.5] * nG` will be `[6.5, 6.5]`, and `gi`, `gj` will be `[int(6.5), int(6.5)] = [6, 6]`, which means that the center of the object will be in grid cell `[6, 6]`

**Comment by ganLover** [comment]
> @ganLover `x`, `y` (bbox center) in the target data are normalized [here](https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/utils/datasets.py#L113), and will be in the range `[0, 1]` in `build_targets`, so we multiply it with the grid size to find out in what grid the bounding box center is positioned, e.g. if `x`, `y` is `[0.5 0.5]` then `[0.5, 0.5] * nG` will be `[6.5, 6.5]`, and `gi`, `gj` will be `[int(6.5), int(6.5)] = [6, 6]`, which means that the center of the object will be in grid cell `[6, 6]`

Thanks
